enews-node/newsFetcher.js

```diff
@@ -1,72 +1,70 @@
 module.exports = NewsFetcher;
 
+var Slack = require('./slack.js');
+var Q = require('q');
+
 function NewsFetcher(token) {
   const GENERAL_CHANNEL_ID = 'C029J9QTH'
-  const HISTORY_ENDPOINT = 'https://slack.com/api/channels.history'
-  const MESSAGE_COUNT = 1000;
-
-  var httpClient = require('request');
 
-  this.getEnews = function getEnews(oldest, latest, callback) {
-    var allEnews = [];
-    var handler = function(enews, hasMore, reducedLatest) {
-      allEnews = allEnews.concat(enews);
-      if (hasMore) {
-        getSlackHistory(createHistoryRequest(oldest, reducedLatest), handler);
-      } else {
-        callback(allEnews);
-      }
-    }
-    getSlackHistory(createHistoryRequest(oldest, latest), handler);
-  }
+  var slack = new Slack(token);
 
-  function getSlackHistory(request, callback) {
-    httpClient.get(request, function(error, response, body) {
-      var jsonBody = JSON.parse(body);
-      var messages = jsonBody.messages;
-      var enews = parseMessages(messages);
-      var reducedLatest = messages[messages.length - 1].ts;
-      callback(enews, jsonBody.has_more, reducedLatest);
-    });
+  this.getEnews = function(oldest, latest, callback) {
+    var wrap = function(messages) {
+      convertToEnews(messages, callback);
+    };
+    slack.getMessages(GENERAL_CHANNEL_ID, oldest, latest, wrap);
   }
 
-  function parseMessages(messages) {
-    var enews = [];
-    messages.forEach(function(message) {
-      if (isEnewsMessage(message)) {
-        enews.push(message.text);
-      }
+  function convertToEnews(messages, callback) {
+    var eNewsMessages = messages.filter(isEnewsMessage);
+    var usersPromise = getUsersFrom(eNewsMessages);
+    Q.all(usersPromise).then(users => {
+      var eNews = toEnews(eNewsMessages, users);
+      callback(eNews);
     });
-    return enews;
   }
 
   function isEnewsMessage(message) {
-    return isValidMessage(message) && isEnews(message.text)
+    var messageText = message.text;
+    return contains(messageText, '#enews') || contains(messageText, '#eNews') || contains(messageText, '#C0YNBKANM');
```

Comment:
it's the redirect slack creates for `#enews` channel, for now I guess it's a constant